### PR TITLE
Fixes #320

### LINKED ISSUE #320: [Bug]: Fix Integration tests post #304
### Python Version

N/A

### Describe the bug

#304 has reduced the test suite completion time, but has resulted in a higher failure rate of the integration tests. A fix is available in #319 and will be duplicated into this branch to expedite the fix.

### Steps to reproduce the behaviour

See: https://github.com/pybop-team/PyBOP/actions/runs/9019436453/job/24782313287

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BradyPlanden**
# Description

This PR updates the `test_parameterisation.py` integration tests. This is completed by hyper-parameter updates for the extended synthetic data and a change to the parameter priors to remove incorrect biasing to the MAP cost function.

In addition, it adds a prior resampling method to improve pybop.SciPyMinimize stability. This avoids failing test when `allow_infeasible_solutions == False` as x0 will be resampling when possible. To achieve this, the `pybop.Prior` classes have been refactored through the addition of a base class `pybop.BasePrior`.

## Issue reference
Fixes #320 

## Review
Before you mark your PR as ready for review, please ensure that you've considered the following:
- Updated the [CHANGELOG.md](https://github.com/pybop-team/PyBOP/blob/develop/CHANGELOG.md) in reverse chronological order (newest at the top) with a concise description of the changes, including the PR number.
- Noted any breaking changes, including details on how it might impact existing functionality.

## Type of change
- [ ] New Feature: A non-breaking change that adds new functionality.
- [ ] Optimization: A code change that improves performance.
- [ ] Examples: A change to existing or additional examples.
- [ ] Bug Fix: A non-breaking change that addresses an issue.
- [ ] Documentation: Updates to documentation or new documentation for new features.
- [ ] Refactoring: Non-functional changes that improve the codebase.
- [ ] Style: Non-functional changes related to code style (formatting, naming, etc).
- [X] Testing: Additional tests to improve coverage or confirm functionality.
- [ ] Other: (Insert description of change)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybop-team/PyBOP/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All unit tests pass: `$ nox -s tests`
- [ ] The documentation builds: `$ nox -s doctest`

You can run integration tests, unit tests, and doctests together at once, using `$ nox -s quick`.

## Further checks:
- [ ] Code is well-commented, especially in complex or unclear areas.
- [ ] Added tests that prove my fix is effective or that my feature works.
- [ ] Checked that coverage remains or improves, and added tests if necessary to maintain or increase coverage.

Thank you for contributing to our project! Your efforts help us to deliver great software.

### CODE REVIEW

File: pybop/optimisers/scipy_optimisers.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def __init__(self, method=None, bounds=None, maxiter=None, tol=1e-5):
         super().__init__()
         self.method = method
         self.bounds = bounds
+        self.num_resamples = 40
```
Comment (BradyPlanden) [inline pybop/optimisers/scipy_optimisers.py:28]: This line is intentionally left without an set/get as #255 enables kwargs to this class, which will be added once that is merged.

File: tests/unit/test_optimisation.py [hunk 2]
```diff
@@ -142,6 +140,41 @@ def test_prior_sampling(self, cost):
...
+
+        # Create the optimisation class with infeasible solutions disabled
+        opt = pybop.Optimisation(
+            cost=cost, optimiser=pybop.SciPyMinimize, allow_infeasible_solutions=False
+        )
+        opt.set_max_iterations(1)
+
+        # If small sigma, expect a ValueError due inability to resample a non np.inf cost
+        if expect_exception:
+            with pytest.raises(ValueError):
```
Comment (NicolaCourtier) [inline tests/unit/test_optimisation.py:173]: Please clarify what this test checks by adding a `match` argument to `pytest.raises`

**Review by NicolaCourtier — APPROVED** [review]
Thanks @BradyPlanden - I've added a few small comments to be addressed, as well as some notes for myself regarding future updates.

**Comment by NicolaCourtier** [comment]
One more: please add a Changelog entry - thanks!

**Comment by NicolaCourtier** [comment]
Thanks @BradyPlanden ! Let's proceed with the name change and merge as it looks like a significant improvement.

**Pull request merged**